### DISCUSSION TITLE
The

### DISCUSSION THREAD

**Discussion by figuewang** [rabbitmq/rabbitmq-server#9856] [Questions] [answered]
help me !!!
rabbitmq 3.7.6
a slave node, The memory is up 300G In an hour. 
this node send data on port 25672,the data as follows picture:

--- SCREENSHOT REMOVED ---

**Comment by figuewang** [comment]
the node send data to master  all the time, the data below:
the content is "exchange deleted"

--- SCREENSHOT REMOVED ---

**Reply by michaelklishin** [reply]
`exchange_deleted` is [one of the internal events](https://rabbitmq.com/logging.html#internal-events). They are very small, so in order to create a massive backlog of these events you have to either have a really high exchange churn rate, or delete a virtual host with, say, millions of exchanges.

Both are scenarios that I do not recall seeing in practice in about 13 years of my active involvement with RabbitMQ.

So I am not convinced this is the root cause. Instead of guessing, see what `rabbitmq-diagnostics memory_breakdown` and the top Erlang processes in `rabbitmq_top` suggest.

**Reply by michaelklishin** [reply]
That is to say, I am certain that these events are not the key contributing factor. They are a red herring. See above for what we'd recommend using to narrow down the key contributing factor.

**Comment by mkuratczyk** [comment, marked as answer]
RabbitMQ 3.7.6 is over 5 years old and 3.7.x went out of support over 3 years ago. Running this in production is simply professional malpractice at this point.

We have made tremendous improvement in memory usage since then. Quite likely, you will not have this problem if you simply upgrade to a recent version (preferably 3.12).

**Comment by michaelklishin** [comment]
A few things that can be added:

 * We do not guess in this community. RabbitMQ [provides tools that help you reason about memory usage](https://rabbitmq.com/memory-use.html) of a node
 * Port 25672 is used for inter-node communication. Combined with Cowboy exceptions in the logs we can come up with a hypothesis: something causes RabbitMQ nodes to transfer a lot of data between the nodes themselves, and the HTTP API (powered by Cowboy) is involved. There is one very common scenario where this happens: an application calls `GET /api/queues` without any pagination, resulting in enormously large data transfers between nodes (each queue has over 70 metrics, you do the math what the payload may be if there are 100K of them).

**Comment by figuewang** [comment]
dear michaelklishin 
thanks you for your answer，
but i wang to konw ，why   "exchange_deleted" happen.  it is not normal . @michaelklishin

**Comment by figuewang** [comment]
dear michaelklishin
help me delete picture above. @michaelklishin

**Reply by figuewang** [reply]
dear @mkuratczyk 
help me delete picture above

**Reply by mkuratczyk** [reply]
I've removed the screenshots
